### DISCUSSION TITLE
Geolocator.RequestAccessAsync does not prompt the user for access

### DISCUSSION THREAD

**Discussion by Zerachiel01** [unoplatform/uno#18094] [Q&A] [answered]
Hello everyone!

I am trying to get the user's current location on an android device, therefore I added Location to the capabilities in the Package.appmanifest file and then ask for location permissions via `Geolocator.RequestAccessAsync` as mentioned in the [UWP docs](https://learn.microsoft.com/en-us/windows/uwp/maps-and-location/get-location).
Unfortunately, the user does not get prompted to allow access to the device's location. Instead, `Geolocator.RequestAccessAsync` just returns a "Declined"-result.
I read about calling `Geolocator.RequestAccessAsync` from the UI-thread, so I called it from several locations in my code like a button click, the OnLaunched event and from a Dispatcher, but nothing worked.
I'm not sure if this is a bug or if I am doing something wrong, so I wanted to ask here first.
So am I doing something wrong, am I missing something? Could someone give me a small snippet on how to do this the right way?

Thanks in advance!

**Comment by morning4coffe-dev** [comment, marked as answer]
Hey @Zerachiel01,

If you are trying to access the location on an Android device, you will need to specify the permissions in the Android manifest (Platforms>Android>`AndroidManifest.xml`). I think the [implementation](https://github.com/unoplatform/uno/blob/0a56fe88d7b4179f0b6623cef9a8bf53e07d9d11/src/Uno.UWP/Devices/Geolocation/Geolocator.Android.cs#L129) checks for Fine Location permission, so you should include `<uses-permission android:name="android.permission.ACCESS_FINE_LOCATION" />`. You should also make sure you have Location enabled on the device. 
Let me know if this helps.

**Reply by Zerachiel01** [reply]
Now it's working, thanks a lot!

**Reply by jeromelaban** [reply]
Looking at our documentation (https://platform.uno/docs/articles/features/windows-devices-geolocation.html), it seems an android section would benefit from this specific configuration. Could you take a look @morning4coffe-dev ? Thanks!
